packages/@aws-cdk/aws-codebuild/lib/source.ts

```diff
@@ -18,67 +49,54 @@ export interface BuildSourceProps {
 /**
  * Source provider definition for a CodeBuild Project.
  */
-export abstract class BuildSource {
+export abstract class Source implements ISource {
+  public static s3(props: S3SourceProps): S3Source {
+    return new S3Source(props);
+  }
+
+  public static codeCommit(props: CodeCommitSourceProps): CodeCommitSource {
+    return new CodeCommitSource(props);
+  }
+
+  public static gitHub(props: GitHubSourceProps): GitHubSource {
+    return new GitHubSource(props);
+  }
+
+  public static gitHubEnterprise(props: GitHubEnterpriseSourceProps): GitHubEnterpriseSource {
+    return new GitHubEnterpriseSource(props);
+  }
+
+  public static bitBucket(props: BitBucketSourceProps): BitBucketSource {
+    return new BitBucketSource(props);
```

Comment:
Nice... Let's make all the classes internal

Reply:
Done.